programs/vote/src/vote_state.rs

```diff
@@ -161,16 +175,30 @@ pub struct VoteState {
 
 impl VoteState {
     pub fn new(vote_init: &VoteInit, clock: &Clock) -> Self {
+        let mut authorized_voter = BTreeMap::new();
+        authorized_voter.insert(clock.epoch, vote_init.authorized_voter);
         Self {
             node_pubkey: vote_init.node_pubkey,
-            authorized_voter: vote_init.authorized_voter,
-            authorized_voter_epoch: clock.epoch,
+            authorized_voter,
             authorized_withdrawer: vote_init.authorized_withdrawer,
             commission: vote_init.commission,
             ..VoteState::default()
         }
     }
 
+    pub fn get_authorized_voter(&self, epoch: u64) -> Option<Pubkey> {
```

Comment:
In the default case, If you use the initial signer as the authorized voter then you might be able to get rid of the option?

Reply:
Hm nvm seems like empty is used for other checks elsewhere.

Reply:
And this api takes an epoch argument. Totally missed that.